rt/rs/extensions/rx2/src/main/java/org/apache/cxf/jaxrs/rx2/client/FlowableRxInvokerImpl.java

```diff
@@ -147,34 +150,90 @@ public FlowableRxInvokerImpl(WebClient wc, ExecutorService ex) {
 
     @Override
     public <T> Flowable<T> method(String name, Entity<?> entity, Class<T> responseType) {
+        
+        Flowable<T> flowable = Flowable.create(new FlowableOnSubscribe<T>() {
+            @Override
+            public void subscribe(FlowableEmitter<T> emitter) throws Exception {
+                    try {
+                        T response = syncInvoker.method(name, entity, responseType);
+                        emitter.onNext(response);
+                        emitter.onComplete();
+                    } catch (Throwable e) {
+                        emitter.onError(e);
+                    }
+            }
+        }, BackpressureStrategy.DROP);
+        
         if (sc == null) {
-            return Flowable.fromFuture(wc.async().method(name, entity, responseType));
+            return flowable;
```

Comment:
This part is concerning. I do understand we couldn't do much if `executorService` is not provided. But it means the flow is going to be synchronous (whereas `RxInvoker` states explicitly in javadocs `... asynchronous computation ...`). May be we could default the executor to `Schedulers.io()` in this case?

Reply:
Just to give a reasoning behind it: the sensible default (like `Schedulers.io()` in this case) is used in many part of Java standard library (notably, `ForkJoin.commonPool()`): when the executor is not provided, the default one is used. Not the perfect choice but the asynchronous nature of the call is preserved.

Reply:
Agree.  Thanks for the comments.  Done.